### PR TITLE
Add simple sinatra example

### PR DESCRIPTION
**Pull request by sbauch**
I'm working on OSS for the other side of SAML and a simple / faked IDP is really useful for our CI, QA and demo purposes - thanks so much for this gem!

I used this to to hack together a [sinatra app](https://github.com/enterprise-oss/sinatra-ruby-idp) and thought I might add back a simple example for getting that working.

I'd love to figure out how to use the views from the gem rather than my recreated ones. I think also this gem might need an `xmlenc` runtime dependency?

I'm using a deployed version of the sinatra app in the flow described here - https://github.com/nextauthjs/next-auth/pull/928

### CODE REVIEW

**Comment by mjobin-mdsol** [comment]
there is an open PR already to add the `xmlenc` gem as a runtime dependency.
it was initially decided to have that optional since not everybody needs/wants it.

**Comment by mjobin-mdsol** [comment]
thanks for the sinatra example

**Comment by Zogoo** [comment]
@sbauch Good job! 
But, one small request that I have. Can you add very simple test that prove your example app will work with gem itself. Because, for future maintenance we might need to change or update gems. And if we don't have a test we might need to ask manual test all that kind of changes.

**Comment by Zogoo** [comment]
@mjobin-mdsol about xmlenc.
I think gem is required to into Gem spec's gem because, this gem has part of logic that actually using xmlenc.

File: examples/sinatra/spec/app_spec.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+describe 'POST /saml-login ' do
+  it 'encodes a SAMLResponse' do
+    allow_any_instance_of(SamlIdp::Controller).to receive(:decode_request).and_return(true)
+    allow_any_instance_of(SamlIdp::Controller).to receive(:encode_response).and_return("SAMLResponse")
+
+    post("/saml-login")
+
+    expect(last_response).to be_ok
+  end
+end
```
Comment (sbauch) [inline examples/sinatra/spec/app_spec.rb:14]: here's a simple test @Zogoo - i dont want to test the gem's internals, but this seems to be to be a pointless test.

happy to iterate to your liking but im not sure what you are hoping to accomplish here? I don't think it makes sense to use a real SAMLRequest (that would be testing the library).

**Comment by mjobin-mdsol** [comment]
I was thinking the exact same thing as @Zogoo when revisiting this. If you could add a test that runs your Sinatra example, that would be super awesome

About `xmlenc` @Zogoo we can talk about it on #128 but it is only required if you add the `encryption: {}` option hash to `encode_response` call in your POST response. which is the default in the readme, thus the confusion.

File: examples/sinatra/views/saml_post.erb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  <head>
+    <meta charset="utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
+  </head>
+  <body onload="document.forms[0].submit();" style="visibility:hidden;">
+    <% form_tag(saml_acs_url) do %>
+      <%= hidden_field_tag("SAMLResponse", { value: @saml_response }) %>
+      <%= hidden_field_tag("RelayState", { value: params[:RelayState] }) %>
+      <%= submit_tag "Submit" %>
+    <% end %>
```
Comment (mjobin-mdsol) [inline examples/sinatra/views/saml_post.erb:12]: why would the post view be such a form?

File: examples/sinatra/views/saml_post.erb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
+  </head>
```
Comment (mjobin-mdsol) [inline examples/sinatra/views/saml_post.erb:6]: why isn't the html header required in the `login.erb` view but you need it here?

**Review by mjobin-mdsol — APPROVED** [review]

**Comment by sbauch** [comment]
I'm happy to write a better test, but not sure what you all are looking for beyond what I added.

Maybe a quick little pseudocode to show what you'd want?

File: examples/sinatra/spec/app_spec.rb (hunk 1 above)
Comment (Zogoo) [reply examples/sinatra/spec/app_spec.rb:14]: I was expecting more working example that proof this Sinatra example work with our future changes.
I'm afraid that if we stub responses and if we merge some pull request that breaks your "sample" then your sample no longer gets useful for others.
So, could you please add one proper response test for your sample app?

**Pull request closed without merging**